src/common/copilot/PowerPagesCopilot.ts

```diff
@@ -45,13 +111,182 @@ export class PowerPagesCopilot implements vscode.WebviewViewProvider {
+            CESUserFeedback(this._extensionContext, sessionID, userID, "thumbsDown", this.telemetry)
+          }
+        }
+      }
+    });
+  }
+
+  public show() {
+    if (this._view) {
+      // Show the webview view
+      this._view.show(true);
+    }
+  }
+
+  private async handleLogin() {
 
+    const pacOutput = await this._pacWrapper.activeOrg();
+    if (pacOutput.Status === "Success") {
+      this.handleOrgChangeSuccess.call(this, pacOutput);
+
+      intelligenceAPIAuthentication().then(({ accessToken, user }) => {
+        this.intelligenceAPIAuthenticationHandler.call(this, accessToken, user);
+      });
+
+    } else if (this._view?.visible) {
+
+      const userOrgUrl = await showInputBoxAndGetOrgUrl();
+      if (!userOrgUrl) {
+        userName = "";
+        this.sendMessageToWebview({ type: 'userName', value: userName });
+
+        if (!this.loginButtonRendered) {
+          this.sendMessageToWebview({ type: "welcomeScreen" });
+          this.loginButtonRendered = true; // Set the flag to indicate that the login button has been rendered
+        }
+
+        return;
+      }
+      const pacAuthCreateOutput = await this._pacWrapper.authCreateNewAuthProfileForOrg(userOrgUrl);
+      if (pacAuthCreateOutput.Status === "Success") {
+        intelligenceAPIAuthentication().then(({ accessToken, user }) => {
+          this.intelligenceAPIAuthenticationHandler.call(this, accessToken, user);
+        });
+      } else {
+        vscode.window.showErrorMessage("Error creating auth profile for org");
+      }
+
+    }
+  }
+
+  private async handleOrgChangeSuccess(pacOutput: PacActiveOrgListOutput) {
+    const activeOrg = pacOutput.Results;
+
+    orgID = activeOrg.OrgId;
+
+    environmentName = activeOrg.FriendlyName;
+
+    userID = activeOrg.UserId;
+
+    activeOrgUrl = activeOrg.OrgUrl;
+
+    showConnectedOrgMessage(environmentName, activeOrgUrl);
   }
 
+  private async intelligenceAPIAuthenticationHandler(accessToken: string, user: string) {
+    if (accessToken && user) {
+
+      apiToken = accessToken;
+      userName = getUserName(user);
+      this.sendMessageToWebview({ type: 'userName', value: userName });
+      this.sendMessageToWebview({ type: "welcomeScreen" });
+    }
+  }
+
+
+
+  private getActiveEditorContent(): { activeFileParams: string[], activeFileContent: string } {
+    const activeEditor = vscode.window.activeTextEditor;
+    if (activeEditor) {
+      const document = activeEditor.document;
+      const fileName = document.fileName;
+      const relativeFileName = vscode.workspace.asRelativePath(fileName);
+
+      const activeFileParams: string[] = getLastThreeParts(relativeFileName);
+
+      const activeFileParamsMapped = this.getMappedParams(activeFileParams);
+
+      return { activeFileParams: activeFileParamsMapped, activeFileContent: document.getText() };
+    }
+    return { activeFileParams: ['', '', ''], activeFileContent: "" };
```

Comment:
check how to assign default values. This does not look scalable. If I add another property need to update the value here too

Reply:
done